marlowe-dashboard-client/src/Capability/PlutusApps/MarloweApp.purs

```diff
@@ -51,20 +57,36 @@ class MarloweApp m where
     :: PlutusAppId
     -> Map TokenName Address
     -> Contract
-    -> m (AjaxResponse (UUID /\ Aff (Either MarloweError MarloweParams)))
+    -> m
+         ( AjaxResponse
+             (UUID /\ Aff (Maybe (Either MarloweError MarloweParams)))
+         )
   applyInputs
     :: PlutusAppId
     -> MarloweParams
     -> TransactionInput
-    -> m (AjaxResponse (Aff (Either MarloweError Unit)))
+    -> m (AjaxResponse (Aff (Maybe (Either MarloweError Unit))))
```

Comment:
This is a weird one, why Maybe Either? What is the semantic meaning of this? Can't we create a new data type that wraps MarloweError and whatever this Maybe should represent?

Reply:
Yeah, that's probably a good idea at this point... I didn't want to incur the overhead of copying all the instance boilerplate to a newtype...

The meaning is:

```
Nothing: The request timed out
Just (Left err): The request completed with an error
Just (Right a): The request completed successfully
```

Reply:
I'll address this on my local branch, I'm editing these files there as well